test/iterators.jl

```diff
@@ -1206,3 +1204,18 @@ end
 @test Base.infer_return_type((Vector{Any},)) do xs
     [x for x in xs if x isa Int]
 end == Vector{Int}
+
+@testset "issue #58922" begin
+    # `last` short circuits correctly
+    @test last(zip(1:10, 2:11)) == (10, 11)  # same length
+    @test last(zip(1:3, 2:11)) == (3, 4)     # different length
+
+    # Finite-guarded zip iterator: one iterator bounded and the other is not
+    @test last(zip(1:3, Iterators.countfrom(2))) == (3, 4)
+    @test last(zip(1:3, Iterators.cycle(('x', 'y')))) == (3, 'x')
+    @test last(zip(1:3, Iterators.repeated('x'))) == (3, 'x')
+
+    # TODO: test for other array types
```

Comment:
```
isdefined(Main, :OffsetArrays) || @eval Main include("testhelpers/OffsetArrays.jl")
using .Main.OffsetArrays
```

is a pattern used in many test suites to get `OffsetArrays` in Base tests (put at the top

Reply:
Resolved in 86d2f58 (rebased to keep history clean)